### PR TITLE
PWN-5812 - Refactor moonpay history transactions

### PR DESCRIPTION
**Pull request by Davran331122**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-5812

### CODE REVIEW

**Review by 1sakov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/moonpay/serversideapi/response/SellTransactionStatus.kt [hunk 1]
```diff
@@ -10,6 +10,7 @@ enum class SellTransactionStatus {
     PENDING,
 
     @SerializedName("failed")
+
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/moonpay/serversideapi/response/SellTransactionStatus.kt:13]:
```suggestion
```

File: app/src/main/java/org/p2p/wallet/moonpay/ui/transaction/SellTransactionDetailsPresenter.kt [hunk 2]
```diff
@@ -205,16 +203,42 @@ class SellTransactionDetailsPresenter(
...
                 tokenToSend = solToken,
-                sendAmount = currentTransaction.formattedSolAmount.toBigDecimal(),
-                receiverAddress = currentTransaction.receiverAddress
+                sendAmount = transaction.formattedSolAmount.toBigDecimal(),
+                receiverAddress = transaction.receiverAddress
             )
         }
     }
 
+    override fun load(transactionId: String) {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/moonpay/ui/transaction/SellTransactionDetailsPresenter.kt:215]: to the up, near the fields
this method called first, it should be placed first

File: app/src/main/java/org/p2p/wallet/moonpay/ui/transaction/SellTransactionDetailsPresenter.kt [hunk 3]
```diff
@@ -205,16 +203,42 @@ class SellTransactionDetailsPresenter(
...
+        launch {
+            try {
+                val sellTransaction = historyInteractor.findTransactionById(transactionId)
+                if (sellTransaction is SellTransaction) {
+                    val currentTransaction = mapper.toSellDetailsModel(sellTransaction)
+                    val viewState = when (currentTransaction.status) {
+                        SellTransactionStatus.WAITING_FOR_DEPOSIT -> {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/moonpay/ui/transaction/SellTransactionDetailsPresenter.kt:222]: put {} everywhere, or nowhere

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/history/ui/historylist/HistoryListViewPresenter.kt [hunk 4]
```diff
@@ -66,7 +65,22 @@ class HistoryListViewPresenter(
         }
     }
 
-    override fun refreshHistory() = Unit
+    override fun refreshHistory() {
+        launch {
+            try {
+                delay(300L)
```
[the remark this answers is not in the document]
Comment (Davran331122) [reply app/src/main/java/org/p2p/wallet/history/ui/historylist/HistoryListViewPresenter.kt:71]: oh, i just copied

**Pull request merged**